src/libmongoc/src/mongoc/mongoc-error.c

```diff
@@ -226,3 +224,84 @@ _mongoc_error_copy_labels_and_upsert (const bson_t *src,
 
    bson_append_array_end (dst, &dst_labels);
 }
+
+/* Defined in SDAM spec under "Application Errors".
+ * @error should have been obtained from a command reply, e.g. with
+ * _mongoc_cmd_check_ok.
+ */
+bool
+_mongoc_error_is_shutdown (bson_error_t *error)
+{
+   if (error->domain != MONGOC_ERROR_SERVER &&
+       error->domain != MONGOC_ERROR_WRITE_CONCERN) {
+      return false;
+   }
+   switch (error->code) {
+   case 11600: /* InterruptedAtShutdown */
+   case 91:    /* ShutdownInProgress */
+      return true;
+   default:
+      return false;
+   }
+}
+
+bool
+_mongoc_error_is_not_master (bson_error_t *error)
+{
+   if (error->domain != MONGOC_ERROR_SERVER &&
+       error->domain != MONGOC_ERROR_WRITE_CONCERN) {
+      return false;
+   }
+   switch (error->code) {
+   case 10107: /* NotMaster */
+   case 13435: /* NotMasterNoSlaveOk */
+      return true;
+   default:
+      return NULL != strstr (error->message, "not master");
+   }
+}
```

Comment:
Hoho, good catch. I think that's in there because of the substring test. The error message
"not master or secondary" is a "node is recovering" error, but not a "not master" error I suppose.

Behaviorally, that means a single threaded client could be marking the topology as stale for a "node is recovering" error when it shouldn't be.